go/kbfs/libkbfs/folder_branch_ops.go

```diff
@@ -3558,7 +3559,24 @@ func (fbo *folderBranchOps) finalizeMDWriteLocked(ctx context.Context,
 		}
 	}
 
-	md.loadCachedBlockChanges(ctx, bps, fbo.log)
+	mdCopy, err := md.deepCopy(fbo.config.Codec())
```

Comment:
You can avoid the whole copy if `md.data.Changes.Ops != nil` already.  That's already checked in `loadCachedBlockChanges`, so in fact it might make more sense to change `loadCachedBlockChanges` itself to make and return a copy, if a copy is needed?  Since there are a bunch more callers of this function in this file.

I.e., `md.loadCachedBlockChanges` returns itself if `md.data.Changes.Ops != nil`, otherwise it returns a copy of itself with the changes made.  (It would also need to accept the codec as an argument.)  If it also returned a bool to say whether a copy was made, then all the callers could make a new `ImmutableRootMetadata` in that case.